spark/src/main/scala/geotrellis/spark/io/hadoop/KryoRegistrator.scala

```diff
@@ -70,6 +75,207 @@ class KryoRegistrator extends SparkKryoRegistrator {
+    kryo.register(classOf[org.osgeo.proj4j.units.Unit])
+    kryo.register(classOf[scala.collection.mutable.WrappedArray$ofInt])
+    kryo.register(classOf[scala.collection.mutable.WrappedArray$ofRef])
+    kryo.register(classOf[scala.collection.Seq[Any]])
+    kryo.register(classOf[scala.Tuple3[Any,Any,Any]])
+
+    /* Special Handling: Problematic Classes */
+    List(
+      "geotrellis.proj4.LatLng$",
+      "geotrellis.spark.EmptyBounds$",
+      "scala.collection.immutable.Nil$",
+      "scala.math.Ordering$Double$",
+      "scala.math.Ordering$Float$",
+      "scala.math.Ordering$Int$",
+      "scala.math.Ordering$Long$",
+      "scala.None$"
+    ).foreach({ c => kryo.register(java.lang.Class.forName(c)) })
+
+    /* Special Handling: Avro */
+    {
+      val booleanSchema = Schema.create(Schema.Type.BOOLEAN)
+      val bytesSchema = Schema.create(Schema.Type.BYTES)
+      val doubleSchema = Schema.create(Schema.Type.DOUBLE)
+      val floatSchema = Schema.create(Schema.Type.FLOAT)
+      val intSchema = Schema.create(Schema.Type.INT)
+      val longSchema = Schema.create(Schema.Type.LONG)
+      val nullSchema = Schema.create(Schema.Type.NULL)
+      val stringSchema = Schema.create(Schema.Type.STRING)
+      val fields = new java.util.ArrayList[Schema.Field]
+      val field = new Schema.Field("a", Schema.create(Schema.Type.NULL), null, null); fields.add(field)
+      val schemas = new java.util.ArrayList[Schema]; schemas.add(booleanSchema); schemas.add(bytesSchema)
+      val enums = new java.util.ArrayList[String]; enums.add("b"); enums.add("c")
+
+      kryo.register(field.getClass)       // Field
+      kryo.register(field.order.getClass) // Field$order
+      kryo.register(classOf[Schema.Type]) // Schema$Type
+
+      kryo.register(booleanSchema.getClass) // Boolean
+      kryo.register(bytesSchema.getClass)   // Bytes
+      kryo.register(doubleSchema.getClass)  // Double
+      kryo.register(floatSchema.getClass)   // Float
+      kryo.register(intSchema.getClass)     // Int
+      kryo.register(longSchema.getClass)    // Long
+      kryo.register(nullSchema.getClass)    // Null
+      kryo.register(stringSchema.getClass)  // String
+
+      kryo.register(Schema.createArray(nullSchema).getClass)          // Array
+      kryo.register(Schema.createEnum("d", "e", "f", enums).getClass) // Enum
+      kryo.register(Schema.createFixed("g", "h", "i", 1).getClass)    // Fixed
+      kryo.register(Schema.createMap(nullSchema).getClass)            // Map
+      kryo.register(Schema.createRecord(fields).getClass)             // Record
+      kryo.register(Schema.createUnion(schemas).getClass)             // Union
+
+      kryo.register(java.lang.Class.forName("org.apache.avro.Schema$LockableArrayList"))
+      kryo.register(java.lang.Class.forName("org.apache.avro.Schema$Name"))
+    }
+
+    val reqRegistration = Properties.envOrElse("GEOTRELLIS_KRYO_REGREQ", null)
+
+    if (reqRegistration != null) {
+      List(
```

Comment:
Ideally.  The issue was that it wasn't clear how to get access to the `kryo` variable from there.  I can investigate this further.